torch_geometric/utils/augmentation.py

```diff
@@ -0,0 +1,190 @@
+        >>> x, node_perm = shuffle_node(x)
+        >>> x
+        tensor([[6., 7., 8.],
+                [3., 4., 5.],
+                [0., 1., 2.]])
+        >>> node_perm
+        tensor([2, 1, 0])
+    """
+    if not training:
+        perm = torch.arange(x.size(0))
+        return x, perm
+    perm = torch.randperm(x.size(0))
+    return x[perm], perm
+
+
+def mask_feature(x: Tensor, p: float = 0.5, column_wise: bool = True,
+                 training: bool = True) -> Tuple[Tensor, Tensor]:
+    r"""Randomly masks feature from the feature matrix
+    :obj:`x` with probability :obj:`p` using samples from
+    a Bernoulli distribution.
+
+    The method returns (1) the retained :obj:`x`, (2) the feature
+    mask indicating where features are retained.
+
+    Args:
+        x (FloatTensor): The feature matrix.
+        p (float, optional): Masking probability. (default: :obj:`0.5`)
+        column_wise (bool, optional): If set to :obj:`True`, this operation
+            mask columns of the feature. Otherwise, the masked features
+            are sampled uniformly. (default: :obj:`True`)
+        training (bool, optional): If set to :obj:`False`, this operation is a
+            no-op. (default: :obj:`True`)
+
+    :rtype: (:class:`FloatTensor`, :class:`BoolTensor`)
+
+    Examples:
+
+        >>> # Masked features are column-wise sampled
+        >>> x = torch.tensor([[0, 1, 2],
+        ...                   [3, 4, 5],
+        ...                   [6, 7, 8]], dtype=torch.float)
+        >>> x, feat_mask = mask_feature(x)
+        >>> x
+        tensor([[0., 0., 2.],
+                [3., 0., 5.],
+                [6., 0., 8.]]),
+        >>> feat_mask
+        tensor([[ True, False,  True],
+                [ True, False,  True],
+                [ True, False,  True]])
+
+        >>> # Masked features are sampled uniformly
+        >>> x, feat_mask = mask_feature(x, column_wise=False)
+        >>> x
+        tensor([[0., 0., 0.],
+                [3., 0., 5.],
+                [0., 0., 8.]])
+        >>> feat_mask
+        tensor([[False, False, False],
+                [ True, False,  True],
+                [False, False,  True]])
+    """
+    if p < 0. or p > 1.:
+        raise ValueError(f'Masking ratio has to be between 0 and 1 '
+                         f'(got {p}')
+    if not training or p == 0.0:
+        return x, torch.ones_like(x, dtype=torch.bool)
+
+    if column_wise:
+        mask = torch.rand(x.size(1), device=x.device) >= p
+        mask = mask.tile(x.size(0), 1)
+    else:
+        mask = torch.randn_like(x) >= p
+
+    # To avoid inplace operation
+    x = x * (mask.float())
+    return x, mask
+
+
+def add_random_edge(edge_index, p: float = 0.2, force_undirected: bool = False,
+                    num_nodes: Optional[Union[Tuple[int], int]] = None,
+                    training: bool = True) -> Tuple[Tensor, Tensor]:
+    r"""Randomly adds edges of ratio :obj:`p` into the existing edges
+    :obj:`edge_index`.
```

Comment:
```suggestion
    r"""Randomly adds edges  to :obj:`edge_index`.
```
Nitpick: maybe more clearer.